operators/src/processing/point_in_polygon.rs

```diff
@@ -0,0 +1,775 @@
+            .iter()
+            .tuple_windows()
+            .zip(time_intervals)
+            .map(
+                move |(
+                    (&multi_polygon_start_index, &multi_polygon_end_index),
+                    multi_polygon_time_interval,
+                )| {
+                    if !multi_polygon_time_interval.intersects(time_interval) {
+                        return false;
+                    }
+
+                    let (multi_polygon_start_index, multi_polygon_end_index) = (
+                        multi_polygon_start_index as usize,
+                        multi_polygon_end_index as usize,
+                    );
+
+                    let mut is_coordinate_in_multi_polygon = false;
+
+                    for (&polygon_start_index, &polygon_end_index) in polygon_offsets
+                        [multi_polygon_start_index..=multi_polygon_end_index]
+                        .iter()
+                        .tuple_windows()
+                    {
+                        let polygon_start_index = polygon_start_index as usize;
+                        let polygon_end_index = polygon_end_index as usize;
+
+                        let mut is_coordinate_in_polygon = true;
+
+                        for (ring_number, (&ring_start_index, &ring_end_index)) in ring_offsets
+                            [polygon_start_index..=polygon_end_index]
+                            .iter()
+                            .tuple_windows()
+                            .enumerate()
+                        {
+                            let is_coordinate_in_ring = self.is_coordinate_in_ring(
+                                coordinate,
+                                ring_start_index as usize,
+                                ring_end_index as usize,
+                            );
+
+                            if (ring_number == 0 && !is_coordinate_in_ring)
+                                || (ring_number > 0 && is_coordinate_in_ring)
+                            {
+                                // coordinate is either "not in outer ring" or "in inner ring"
+                                is_coordinate_in_polygon = false;
+                                break;
+                            }
+                        }
+
+                        if is_coordinate_in_polygon {
+                            is_coordinate_in_multi_polygon = true;
+                            break;
+                        }
+                    }
+
+                    is_coordinate_in_multi_polygon
+                },
+            )
+    }
+
+    /// Is the coordinate contained in any polygon of the collection?
+    ///
+    /// The function returns `true` if the `Coordinate2D` is inside the multi polygon, or
+    /// `false` if it is not. If the point is exactly on the edge of the polygon,
+    /// then the function may return `true` or `false`.
+    ///
+    /// TODO: check boundary conditions separately
+    ///
+    pub fn is_coordinate_in_collection(
```

Comment:
maybe rename `is_coordinate_in_any_polygon`. not super important though

Reply:
65ee56c